src/empty-states/src/EmptyState.js

```diff
@@ -0,0 +1,184 @@
+      justifyContent="center"
+    >
+      <Pane display="flex" alignItems="flex-start" flex={1} height="100%">
+        <Pane paddingRight={majorScale(6)}>
+          <Pane
+            display="flex"
+            justifyContent="center"
+            alignItems="center"
+            borderRadius="50%"
+            backgroundColor={imageBgColor}
+            width={majorScale(9)}
+            height={majorScale(9)}
+          >
+            {React.cloneElement(image, { size: majorScale(4) })}
+          </Pane>
+        </Pane>
+        <Pane display="flex" flexDirection="column" paddingRight={majorScale(6)}>
+          <Heading size={500}>{title}</Heading>
+          {description && (
+            <Paragraph color="muted" marginTop={majorScale(2)}>
+              {description}
+            </Paragraph>
+          )}
+          {hasFooter && (
+            <Pane marginTop={majorScale(5)} display="flex">
+              {primaryCta && React.cloneElement(primaryCta, { marginRight: majorScale(2) })}
+              {secondaryCta}
+              {(primaryCta || secondaryCta) &&
+                anchorCta &&
+                React.cloneElement(anchorCta, { marginLeft: majorScale(2) })}
+              {!primaryCta && !secondaryCta && anchorCta}
+            </Pane>
+          )}
+        </Pane>
+      </Pane>
+    </Pane>
+  )
+}
+
+/* eslint-enable react/prop-types */
+
+/* eslint-disable react/prop-types */
+
+const VerticalOrientation = ({ background, description, image, imageBgColor, primaryCta, title }) => {
+  const { colors } = useTheme()
+  const backgroundColor = background === 'light' ? 'white' : colors.gray75
+
+  return (
+    <Pane
+      display="flex"
+      alignItems="center"
+      flexDirection="column"
+      justifyContent="center"
+      flex={1}
+      backgroundColor={backgroundColor}
+      paddingX={majorScale(5)}
+      paddingY={majorScale(5)}
+      height="100%"
+      width="100%"
+    >
+      <Pane
+        display="flex"
+        justifyContent="center"
+        alignItems="center"
+        borderRadius="50%"
+        backgroundColor={imageBgColor}
+        width={majorScale(7)}
+        height={majorScale(7)}
+      >
+        {React.cloneElement(image, { size: majorScale(3) })}
+      </Pane>
+      <Heading marginTop={majorScale(2)} textAlign="center">
+        {title}
+      </Heading>
+      <Paragraph marginTop={majorScale(2)} textAlign="center">
+        {description}
+      </Paragraph>
+      {primaryCta && React.cloneElement(primaryCta, { marginTop: minorScale(5) })}
+    </Pane>
+  )
+}
+
+/* eslint-enable react/prop-types */
+
+const PrimaryButton = props => {
+  return <Button appearance="primary" {...props} />
+}
+
+const SecondaryButton = props => {
+  return <Button appearance="minimal" {...props} />
+}
+
+const LinkButton = props => {
+  return <Link {...props} target="_blank" size={300} lineHeight="34px" />
+}
+
+const EmptyState = ({
+  anchorCta,
+  background = 'light',
+  description,
+  image,
```

Comment:
One thing that I'm thinking about...do we know / can we guarantee that the icon color and iconbackground color can be compatible? This puts responsibility on consumers to "do the right thing", but I'm just wondering if there's a way we can restrict this API up front

Reply:
Yeah agree that this is putting the responsibility to the consumers. A few ideas:
- Restricted the icon color & background color with intent (success, warn, danger, info) and only allow passing in icon shape (class) and its intent. Not sure if this would be too restricted.
- Restricted to just icon and its color, and we always apply the color minus 200 (red-500 -> red-300) as its background so we get one less prop. This can result in less optimized contrast since our color system was not great.
- Keep as is.

What do you think?

Reply:
Hm..yeah I say we can keep as-is for now and maybe work to make the API more restrictive in the future.